### ISSUE TITLE
prefer-replace-all: Could it handle regexp with the 'u' flag?

### ISSUE THREAD

**Issue by coreyfarrell** [sindresorhus/eslint-plugin-unicorn#511]
```js
string.replace(/test/gu, 'TEST')
```

Is it for technical reasons that this cannot be reported (ie regexp-tree failing to parse it)?  If not it would be nice if use of the unicode flag did not bypass this rule.  The primary use case is for code that has already been updated to comply with `require-unicode-regexp`.

**Comment by sindresorhus** [comment]
// @jmoore914

**Comment by jmoore914** [comment]
Yes, there is no reason that this can't be updated to include the 'u' flag. If a unicode pattern was used (e.g. /\p{L}/gu) then the function `isLiteralCharactersOnly` would return false anyway. This is a quick fix so I should be able to put in a PR for it this weekend.
